commit 2fbeb8d3ba939ab952c52f389c910a46f44b3094 (refs/changes/22/1522/3)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-02-15T16:39:36-08:00 (2 years, 8 months ago)
    
    DAPI-337 dapi's hard-filter-volumes-from mutates vm.internal_metadata['volumesfrom'] such that multiple alloc passes from CNAPI-671 breaks
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/lib/algorithms/hard-filter-volumes-from.js b/lib/algorithms/hard-filter-volumes-from.js
index 203a2ca..42d5362 100644
--- a/lib/algorithms/hard-filter-volumes-from.js
+++ b/lib/algorithms/hard-filter-volumes-from.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 /*
@@ -112,13 +112,14 @@ loadVmAndFilter(log, servers, requiredVms, getVm, cb)
 	log.debug('opts.getVm set; fetching VMs listed in volumes-from');
 
 	var reasons = {};
+	var remainingVms = requiredVms.slice(); // Get a copy, b/c we mutate it.
 
 	function step() {
-		if (requiredVms.length === 0 || servers.length === 0) {
+		if (remainingVms.length === 0 || servers.length === 0) {
 			return (cb(null, servers.slice(0, 1), reasons));
 		}
 
-		var vmUuid = requiredVms.shift();
+		var vmUuid = remainingVms.shift();
 
 		return getVm(vmUuid, function (err, vm) {
 			if (err) {
